pkg/policies/policies.go

```diff
@@ -122,68 +127,84 @@ func (pv *PolicyVerifier) VerifyMaterial(ctx context.Context, material *v12.Atte
 		}
 
-		// it's expected statements can only be validated by policy of type ATTESTATION
-		if spec.GetSpec().GetType() != v1.CraftingSchema_Material_ATTESTATION {
+		// it's expected statements can only be validated by policies of type ATTESTATION
+		types := getPolicyTypes(policy)
+		if !slices.Contains(types, v1.CraftingSchema_Material_ATTESTATION) {
 			continue
 		}
 
-		// 2. load the policy script (rego)
-		script, err := LoadPolicyScriptFromSpec(spec)
+		// 2. load the policy scripts (rego)
+		scripts, err := LoadPolicyScriptsFromSpec(policy, v1.CraftingSchema_Material_ATTESTATION)
 		if err != nil {
 			return nil, NewPolicyError(err)
 		}
 
-		pv.logger.Info().Msgf("evaluating policy '%s' on attestation", spec.Metadata.Name)
-
 		material, err := protojson.Marshal(statement)
 		if err != nil {
 			return nil, NewPolicyError(err)
 		}
 
-		// 4. verify the policy
-		ng := getPolicyEngine(spec)
-		res, err := ng.Verify(ctx, script, material, getInputArguments(policyAtt.GetWith()))
+		pv.logger.Info().Msgf("evaluating policy '%s' on attestation", policy.Metadata.Name)
+
+		violations, scriptBody, err := pv.executeScript(ctx, policy, scripts, material, policyAtt)
 		if err != nil {
 			return nil, NewPolicyError(err)
 		}
 
-		// We store the body in the attestation unless the policy comes from a remote provider
-		// in which case with the reference it will suffice
 		var body string
 		if !IsProviderScheme(ref.GetName()) {
-			body = base64.StdEncoding.EncodeToString(script.Source)
+			body = scriptBody.String()
 		}
 
 		// 5. Store result in the attestation itself (for the renderer to include them in the predicate)
 		result = append(result, &v12.PolicyEvaluation{
-			Name:            spec.Metadata.Name,
+			Name:            policy.Metadata.Name,
 			Body:            body,
-			Violations:      policyViolationsToAttestationViolations(res),
-			Annotations:     spec.GetMetadata().GetAnnotations(),
-			Description:     spec.GetMetadata().GetDescription(),
+			Violations:      policyViolationsToAttestationViolations(violations),
+			Annotations:     policy.GetMetadata().GetAnnotations(),
+			Description:     policy.GetMetadata().GetDescription(),
 			With:            policyAtt.GetWith(),
-			Type:            spec.GetSpec().Type,
+			Type:            policy.GetSpec().Type,
 			ReferenceName:   ref.Name,
 			ReferenceDigest: ref.Digest["sha256"]})
 	}
 
 	return result, nil
 }
 
+func (pv *PolicyVerifier) executeScript(ctx context.Context, policy *v1.Policy, scripts []*engine.Policy, material []byte, att *v1.PolicyAttachment) ([]*engine.PolicyViolation, *strings.Builder, error) {
+	violations := make([]*engine.PolicyViolation, 0)
+	var body strings.Builder
+
+	for _, script := range scripts {
+		// verify the policy
+		ng := getPolicyEngine(policy)
+		res, err := ng.Verify(ctx, script, material, getInputArguments(att.GetWith()))
+		if err != nil {
+			return nil, nil, NewPolicyError(err)
+		}
+
+		body.WriteString(beginPolicySep)
```

Comment:
should we store it as an array of base64? Would it that be possible? If not that's ok.

Reply:
I've created a new property `sources`